### ISSUE TITLE
3.2.6 Findable Help, confusing use of "page" and "web page"

### ISSUE THREAD

**Issue by WilcoFiers** [w3c/wcag#1347]
The word "page" in this SC, and in the definition of "single page web application" is used to mean two different things. One use of it means "web page" as defined in WCAG, and the other seems to mean something along the line of "view", although it is left open to interpretation what that second "page" is. I think that for this to be interpreted consistently, we need a clear definition.

For example, if a SPA opens a print view, is that a different page that requires this link or info, or is it the same page? What about a modal? It seems to me like most situations in which major pieces of the UI become unavailable potentially are "pages" (meaning views) that would not benefit from having this information directly embedded.

**Comment by scha14** [comment]
> Page/ pages in the understanding document seems to mean web page in all instances. Are there any particular instances that stand out so we can review? 

The SC does not cover print view/ PDF pages as described in the [understanding document](https://www.w3.org/WAI/WCAG22/Understanding/findable-help.html)

> It is not the intent of this Success Criteria to require authors to provide help information on PDFs or other static documents that may be available for viewing/download from the Web pages.

**Comment by alastc** [comment]
Hi @WilcoFiers,

> One use of it means "web page" as defined in WCAG, and the other seems to mean something along the line of "view", although it is left open to interpretation what that second "page" is.

I'm not sure what you mean by 'second page'? It sets the scope to SPAs and sets of web-pages, then (within that scope) talks about "access to at least one option ... on each page:" If you mean the SPA, that then uses the definition we agreed previously.

> For example, if a SPA opens a print view, is that a different page that requires this link or info, or is it the same page?

The definition is "pages obtained from a single URI that provides navigation which changes the meaning of the Web page", so if it opens a print view with no navigation, then no. That's also explained in the Understanding doc.

> It seems to me like most situations in which major pieces of the UI become unavailable potentially are "pages" (meaning views) that would not benefit from having this information directly embedded.

Indeed, and I don't think that fits in the set-of-pages or SPA definition.

**Comment by jake-abma** [comment]
also here I'll paste the text from another comment I made:

So I got a question from my teams:

We have a SPA but of course do the routing ourselves (as lot's of site do)
In this case you do not have one page (or do you?!)

This is not clear as you do not 'get the page' from a server...

**Comment by alastc** [comment]
@jake-abma I think the short answer is that it doesn't matter - if it's an SPA that does routing then it's probably a set of web pages (by our dated definitions).

But it isn't a PDF, which helps!

@WilcoFiers I'm fairly sure you were involved in the discussion that created the SPA defintion:

> Pages obtained from a single URI that provide navigation which changes the meaning of the Web page

Neither print previews or modals provide navigation that changes the meaning of the page (without dismissing those features). Is there a further update needed?

**Comment by scha14** [comment]
@WilcoFiers @jake-abma do we need an update? If not, could we close this issue?

**Comment by scha14** [comment]
Draft response (Same as issue https://github.com/w3c/wcag/issues/1368)

> #1538 PR to add text to understanding document
> 
> For a single page Web application, if the same URI routes to pages or views with different layouts or content, then each of those pages or views, have the supported ways of finding help in the same relative order

**Comment by jake-abma** [comment]
@scha14 for me you can close, Wilco might think otherwise

**Comment by alastc** [comment]
During the call yesterday this was update to:

>A single page Web application (compared to a Web_Page) shows multiple "pages" or views of content at the same URI. If a web application uses different URIs for each view of the content, that is considered multiple web pages because the URI changes.

Hopefully that clarifies it, please re-open if not.

**Comment by jake-abma** [comment]
https://github.com/w3c/wcag/issues/1368#issuecomment-737798886

**Comment by scottaohara** [comment]
This reworded text would allow content to shift around when rendering new spa views (but to end users they could interpret these as new “pages” since they likely don’t know or care about the tech used to deliver the site they’re using) but as long as the developers didn’t change the URI, it would pass the SC. That seems an odd loophole.

**Comment by alastc** [comment]
Hi Scott,

Where the URI changes (regardless of the technology) that's covered by our current definition of web pages. It doesn't really matter whether users interpret them as new pages, if the URI changes it's a new page.

The 'hole' we are trying to cover is where an SPA does load new content without updating the URI. I don't understand what you mean about shifting content around, is that different from "shows multiple "pages" or views of content"?

**Comment by scottaohara** [comment]
Yes, that's what I'm referring to.  A SPA loading new screens/views where a different template has been used.  Content ("help" in this situation) that was found in one location of the previously used template is now located in another (e.g., was in the footer, but is no longer there, having moved to another area of the screen / been removed all together).  The URI doesn't update, but to a user this appears to be a different "page" because they activated a link to get here, and content is completely different from the previous screen.

based on this line:
>If a web application uses different URIs for each view of the content, that is considered multiple web pages because the URI changes.

It reads to me that if the URI doesn't change, then it's not considered different pages.  

I see the [definition for single page web app](https://www.w3.org/TR/WCAG22/#dfn-single-page-web-application) says:
>pages obtained from a single URI that provides navigation which changes the meaning of the Web page

So reading these together, it just seems confusing whether or not an SPA needs the URI to change for the SC to consider it multiple pages.  The definition alone would have seemed to indicate that, but the new text seems to indicate that if the URI doesn't change, then it wouldn't be considered multiple pages.

Apologies if this is clear to others and I'm just mucking up the conversation here.

**Comment by alastc** [comment]
>  The URI doesn't update, but to a user this appears to be a different "page" because they activated a link to get here, and content is completely different from the previous screen.

That's the point of the new definition, it covers that scenario.

So you're correct, if the URI doesn't change it is not considered different pages (by WCAG definition), which is why we have the SPA definition. 

So we have three conditions covered by 2 definitions:

1. Web pages, in the traditional sense.
2. SPAs with different URIs, covered by standard definition of web pages.
3. SPA without routing (same URI), covered by the SPA definition.

If anything the requirement is stricter on the SPA, as for web-pages they have to be part of a 'set of' pages.

**Comment by scottaohara** [comment]
Thanks for outlining that for me @alastc.  i'm glad to read that's the intent of this, to cover the scenario.  for whatever reason, i'm still reading these as being in opposition, rather than implying that they are meant to be adjusted based on situation.   but, i'm more than happy to scamper off and chalk that up to a failing on my part.
